File: hotel-plugin-chat/src/main/java/eu/getsoftware/hotelico/infrastructure/hotel/plugin/chat/adapter/in/web/controller/ChatMSCommunicationController.java
package eu.getsoftware.hotelico.infrastructure.hotel.plugin.chat.adapter.in.web.controller;

import eu.getsoftware.hotelico.clients.api.adapter.in.web.controller.IChatApiController;
import eu.getsoftware.hotelico.clients.api.clients.common.dto.CustomerDTO;
import eu.getsoftware.hotelico.clients.api.clients.infrastructure.chat.dto.ChatMsgDTO;
import eu.getsoftware.hotelico.infrastructure.hotel.plugin.chat.adapter.out.persistence.model.ChatMessageMappedEntity;
import eu.getsoftware.hotelico.infrastructure.hotel.plugin.chat.adapter.out.persistence.model.ChatUserMappedEntity;
import eu.getsoftware.hotelico.infrastructure.hotel.plugin.chat.adapter.out.persistence.repository.ChatMessageRepository;
import eu.getsoftware.hotelico.infrastructure.hotel.plugin.chat.adapter.out.persistence.repository.ChatUserRepository;
import eu.getsoftware.hotelico.infrastructure.hotel.plugin.chat.application.port.out.ChatService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v0/chat")
@RequiredArgsConstructor
public class ChatMSCommunicationController implements IChatApiController {
	
	private final ChatMessageRepository chatMessageRepository;	
	private final ChatUserRepository chatUserRepository;
	private final ChatService chatService;
	
//	@PostMapping("/message")  in interface
	@Override public ChatMsgDTO postMessage(@RequestBody ChatMsgDTO msgDTO) {
		
		ChatMessageMappedEntity createdEntity = chatService.createChatMessageFromDTO(msgDTO);
		return chatService.convertToDTO(createdEntity);
	}
	
	
//	@PostMapping("/customer") in interface
	public CustomerDTO updateUser(@RequestBody CustomerDTO customerDTO) {
		
		ChatUserMappedEntity updateEntity = chatService.updateCustomerFromDTO(customerDTO);
		return chatService.convertToDTO(updateEntity);
	}
}

File: hotel-chat-service/src/main/kotlin/eu/getsoftware/hotelico/chat/adapter/in/web/controller/ChatMSCommunicationController.kt
package eu.getsoftware.hotelico.chat.adapter.`in`.web.controller

//import eu.getsoftware.hotelico.clients.openapi.generated.chat.ChatApi
//import eu.getsoftware.hotelico.clients.openapi.generated.chat.dto.ChatMsgRestDTO
//import eu.getsoftware.hotelico.clients.openapi.generated.chat.dto.CustomerRestDTO
import eu.getsoftware.hotelico.chat.adapter.out.persistence.model.ChatMessageMappedEntity
import eu.getsoftware.hotelico.chat.adapter.out.persistence.model.ChatUserMappedEntity
import eu.getsoftware.hotelico.chat.adapter.out.persistence.repository.ChatMessageRepository
import eu.getsoftware.hotelico.chat.adapter.out.persistence.repository.ChatUserRepository
import eu.getsoftware.hotelico.chat.application.port.out.ChatDTOService
import eu.getsoftware.hotelico.clients.api.application.dto.entity.CustomerDTO
import eu.getsoftware.hotelico.clients.api.application.infrastructure.chat.dto.ChatMsgDTO
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping

@Controller
@RequestMapping("/api/v0/chat")
@RequiredArgsConstructor
class ChatMSCommunicationController(
    private val chatMessageRepository: ChatMessageRepository,
    private val chatUserRepository: ChatUserRepository,
    private val chatDTOService: ChatDTOService
) //: ChatApi 
{

    //	@PostMapping("/message")  in interface
    //override 
    fun postMessage(@RequestBody msgDTO: ChatMsgDTO): ChatMsgDTO {
        val createdEntity: ChatMessageMappedEntity = chatDTOService.createChatMessageFromDTO(msgDTO)
        return chatDTOService.convertToDTO(createdEntity)
    }

    //	@PostMapping("/customer") in interface
    //override 
    fun updateUser(@RequestBody customerDTO: CustomerDTO): CustomerDTO {
        val updateEntity: ChatUserMappedEntity = chatDTOService.updateCustomerFromDTO(customerDTO)
        return chatDTOService.convertToDTO(updateEntity)
    }
}
